### ISSUE TITLE
Add support for type_changing_struct_update

### ISSUE THREAD

**Issue by nazar-pc** [intellij-rust/intellij-rust#9691]
## Environment

* **IntelliJ Rust plugin version:** 0.4.183.4992-223-nightly
* **Rust toolchain version:** 1.65.0-nightly (20ffea693 2022-08-11) x86_64-unknown-linux-gnu
* **IDE name and version:** IntelliJ IDEA 2022.3 Beta Ultimate Edition (IU-223.7401.7)
* **Operating system:** Linux 6.0.7-x64v2-xanmod1
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.cargo.emulate.terminal, org.rust.macros.proc.function-like, org.rust.macros.proc

## Problem description

There is an experimental feature on nightly [type_changing_struct_update](https://github.com/rust-lang/rfcs/blob/master/text/2528-type-changing-struct-update-syntax.md), currently IDE doesn't understand it and produces false-positive inspections.

## Steps to reproduce

```rust
struct S<T> {
    t: T,
    n: u32,
}

let a = S::<u32> { t: 1, n: 1 };
let b = S::<u64> { t: 2, ..a };
```

^ produces following false-positive:
```
mismatched types [E0308] expected `S<u64>`, found `S<u32>
```
